clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/userdata/UserDataTokenizer.java

```diff
@@ -0,0 +1,73 @@
+package com.netflix.spinnaker.clouddriver.aws.deploy.userdata;
+
+import com.netflix.frigga.Names;
+
+/** Utility class to replace tokens in user data templates. */
+public class UserDataTokenizer {
+
+  /**
+   * Replaces the tokens that are present in the supplied user data.
+   *
+   * @param names {@link Names}
+   * @param userDataRequest {@link UserDataProvider.UserDataRequest}
+   * @param rawUserData The user data to replace tokens in
+   * @param legacyUdf
+   * @return String
+   */
+  static String replaceTokens(
+      Names names,
+      UserDataProvider.UserDataRequest userDataRequest,
+      String rawUserData,
+      Boolean legacyUdf) {
+    String stack = isPresent(names.getStack()) ? names.getStack() : "";
+    String cluster = isPresent(names.getCluster()) ? names.getCluster() : "";
+    String revision = isPresent(names.getRevision()) ? names.getRevision() : "";
+    String countries = isPresent(names.getCountries()) ? names.getCountries() : "";
+    String devPhase = isPresent(names.getDevPhase()) ? names.getDevPhase() : "";
+    String hardware = isPresent(names.getHardware()) ? names.getHardware() : "";
+    String zone = isPresent(names.getZone()) ? names.getZone() : "";
+    String detail = isPresent(names.getDetail()) ? names.getDetail() : "";
+
+    // Replace the tokens & return the result
+    String result =
+        rawUserData
+            .replace("%%account%%", userDataRequest.getAccount())
+            .replace("%%accounttype%%", userDataRequest.getAccountType())
+            .replace(
+                "%%env%%",
+                (legacyUdf != null && legacyUdf)
+                    ? userDataRequest.getAccount()
+                    : userDataRequest.getEnvironment())
+            .replace("%%app%%", names.getApp())
+            .replace("%%region%%", userDataRequest.getRegion())
+            .replace("%%group%%", names.getGroup())
+            .replace("%%autogrp%%", names.getGroup())
+            .replace("%%revision%%", revision)
+            .replace("%%countries%%", countries)
+            .replace("%%devPhase%%", devPhase)
+            .replace("%%hardware%%", hardware)
+            .replace("%%zone%%", zone)
+            .replace("%%cluster%%", cluster)
+            .replace("%%stack%%", stack)
+            .replace("%%detail%%", detail)
+            .replace("%%tier%%", "");
+
+    if (userDataRequest.getLaunchTemplate() != null && userDataRequest.getLaunchTemplate()) {
```

Comment:
```suggestion
    if (userDataRequest.getLaunchTemplate() != null) {
```

Odd conditional, isn't the second expression is redundant?

Reply:
In this case `getLaunchTemplate` is a `Boolean` object which can be null and we only want to run this code if it's both not null and true.

Reply:
I didn't realize what value it was. Poor method / property naming, that one. Oh well. Thanks!

Reply:
Yeah that's the default name you get from lombok when you use the `Boolean` object.  👎